web/blog/tenzir-v4.7/index.md

```diff
@@ -0,0 +1,204 @@
+information from a MaxMind GeoIP® database.
+
+To get started, [download the freely available GeoLite2 MaxMind
+database](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data), or use
+any other MaxMind database. We'll use the country database file
+`GeoLite2-Country.mmdb`.
+
+```text {0} title="Create a 'geoip' context named 'country'"
+context create country geoip --db-path /path/to/GeoLite2-Country.mmdb
+```
+
+```text {0} title="Enrich Suricata events with the 'country' context"
+export
+| where #schema == /suricata.*/
+/* Apply the context to both source and destination IP address fields */
+| enrich src_country=country --field src_ip
+| enrich dest_country=country --field dest_ip
+/* Use just the country's isocode, and discard the rest of the information */
+| replace src_country=src_country.context.country.iso_code, 
+          dest_country=dest_country.context.country.iso_code
+```
+
+```json {0} title="Possible output"
+{
+  "timestamp": "2021-11-17T14:02:38.165570",
+  "flow_id": 1837021175481117,
+  "pcap_cnt": 357,
+  "vlan": null,
+  "in_iface": null,
+  "src_ip": "45.137.23.27",
+  "src_port": 47958,
+  "dest_ip": "198.71.247.91",
+  "dest_port": 53,
+  "proto": "UDP",
+  "event_type": "dns",
+  "community_id": "1:0nZC/6S/pr+IceCZ04RjDZbX+KI=",
+  "dns": {
+    // ...
+  },
+  "src_country": "NL",
+  "dest_country": "US"
+}
+```
+
+The [`geoip`](/next/contexts/geoip) context is a powerful building block for
+in-band enrichments. Besides country codes and country names you can add region
+codes, region names, cities, zip codes, and geographic coordinates. With the
+flexibility of the [contextualization framework](/next/contexts) this
+information you can now get this information in real-time.
+
+:::info Follow our Blog Post Series
+Read more about contexts in our blog post series:
+1. [Enrichment Complexity in the Wild](/blog/enrichment-complexity-in-the-wild)
+2. [Contextualization Made Simple](/blog/contextualization-made-simple)
+:::
+
+## Grok and KV Parsers
+
+The [`kv`](/next/formats/kv) and [`grok`](/next/formats/grok) parsers combine
+well with the [`parse`](/next/operators/parse) operator introduced with Tenzir
+v4.6. The former reads key-value pairs by splitting strings based on regular
+expressions, and the latter uses a parser modeled after the [Logstash
+`grok` plugin][logstash-grok] in Elasticsearch.
+
+[logstash-grok]: https://www.elastic.co/guide/en/logstash/current/plugins-filters-grok.html
+
+Parse a fictional HTTP request log with `grok`:
+
+```json {0} title="Example input"
+{
+  "message": "55.3.244.1 GET /index.html 15824 0.043"
+}
+```
+
+```text {0} title="Parse with grok"
+parse message grok "%{IP:client} %{WORD:method} %{URIPATHPARAM:request} %{NUMBER:bytes} %{NUMBER:duration}"
+```
+
+```json {0} title="Example output"
+{
+  "message": {
+    "client": "55.3.244.1",
+    "method": "GET",
+    "request": "/index.html",
+    "bytes": 15824,
+    "duration": "0.043"
```

Comment:
```suggestion
    "duration": 0.043
```

Reply:
Nice catch, thank you! I also updated the image accordingly.